### ISSUE TITLE
1.21 feature_assumevalid.py times out

### ISSUE THREAD

**Issue by rnicoll** [dogecoin/dogecoin#1921]
Opening this to track work on the branch https://github.com/rnicoll/dogecoin/tree/1.21-sync-up-to-auxpow 

The test https://github.com/rnicoll/dogecoin/blob/1.21-sync-up-to-auxpow/test/functional/feature_assumevalid.py (run via https://github.com/rnicoll/dogecoin/blob/1.21-sync-up-to-auxpow/test/functional/test_runner.py ) times out, because the blockchain sync fails. _That_ fails because one of the nodes decides the difficulty bits are incorrect. What I don't know is if the difficulty bits are set correctly and miscalculated on the receiving node, or they're genuinely miscalculated, and in either scenario why. You can see the logs from tests under `/tmp` if you're on Linux.

Probably, somewhere in the like last 20 commits I've changed something I shouldn't; they're basically all cherry-picks from another branch, and one of the conflict resolutions was probably incorrect.

The acceptance criteria for this ticket is to resolve that sync issue and get `feature_assumevalid.py` to pass.

**Comment by kaidhar** [comment]
The timeout is happening because for a minimum difficulty block to mine, the timestamp should be 2*10 minutes apart. But in the test we set self.block_time += 1 ,  editing this  to  self.block_time += 10000, fixes the issues. 

But, assumeValid flag seems to be not working since even for Node1 the max height that is reached is 241.

**Comment by metatron1973** [comment]
Ok who has Bug destroyer  to disrupt any potential bugs ????

**Comment by rnicoll** [comment]
So eventually tracked down the original cause of this; validation.cpp around like 2040 has a check against block time, where a block must be two weeks old for assume valid to work. So when I reduced the block time, the tests 2,000 blocks wasn't enough.

I've rejigged this so on regtest only it will accept like 2 days old, and every other network retains the 2 week requirement (because wall clock time is important for this). That's in #1953 but blocked waiting on earlier PRs to go in.
